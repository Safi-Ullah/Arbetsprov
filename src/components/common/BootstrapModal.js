import React from 'react';
import PropTypes from 'prop-types';

export const BootstrapModal = props => {

    const { handleClose, showModal, heading, children } = props;

    if (showModal)
        return (
            <div className="modal show"
                style={{ display: 'block', paddingRight: '15px'}}
                tabIndex={-1}
                role="dialog"
                aria-hidden="true">
                <div className="modal-dialog modal-dialog-scrollable modal-dialog-centered" role="document">
                    <div className="modal-content">
                        <div className="modal-header">
                            <h5 className="modal-title">{ heading }</h5>
                            <button type="button" className="close" onClick={handleClose}>
                                <span aria-hidden="true">&times;</span>
                            </button>
                        </div>
                        { children }
                    </div>
                </div>
            </div>
        );
    else
        return <div/>;
}

BootstrapModal.propTypes = {
    handleClose: PropTypes.func.isRequired,
    showModal: PropTypes.bool.isRequired,
    heading: PropTypes.string.isRequired,
    children: PropTypes.array
};

BootstrapModal.defaultProps = {
    children: [<div key={1}/>]
};