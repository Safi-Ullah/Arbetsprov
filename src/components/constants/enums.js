export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    }
};