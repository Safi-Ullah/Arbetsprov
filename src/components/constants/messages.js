export const PromptMessages = {
    DELETE_TASK_PROMPT: `Are you sure you want to permanently delete this task?
        You will not be able to restore in the future.`
};