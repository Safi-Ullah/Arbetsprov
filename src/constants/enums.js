export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    },
    APP_SCREENS: {
        dashboard: 1,
        deletedTasks: 2
    },
    SORT_BY_COMPLEXITY: {
        ascending: 1,
        descending: 2
    },
    TEST_TYPES: {
        dom: 'Dom test',
        snapshot: 'Snapshot test'
    }
};