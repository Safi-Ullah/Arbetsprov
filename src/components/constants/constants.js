import { Enums } from './enums';

export const TASK_ACTION_HANDLERS = {
    [Enums.TASK_ACTIONS.edit]: 'displayEditTaskModal',
    [Enums.TASK_ACTIONS.delete]: 'displayConfirmationDialog'
};