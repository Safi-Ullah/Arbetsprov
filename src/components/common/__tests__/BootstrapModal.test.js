import React from 'react';
import TestRenderer from 'react-test-renderer';
import { BootstrapModal } from '../BootstrapModal';
import { Enums } from '../../../constants';

const props = {
    handleClose: jest.fn(),
    showModal: true,
    heading: '',
};

describe(Enums.TEST_TYPES.snapshot, () => {
    it('display modal', () => {
        const tree = TestRenderer.create(<BootstrapModal {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('hidden modal', () => {
        props['showModal'] = false;
        const tree = TestRenderer.create(<BootstrapModal {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});