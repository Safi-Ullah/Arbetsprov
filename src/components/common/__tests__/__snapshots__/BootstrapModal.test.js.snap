// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test display modal 1`] = `
<div
  aria-hidden="true"
  className="modal show"
  role="dialog"
  style={
    Object {
      "display": "block",
      "paddingRight": "15px",
    }
  }
  tabIndex={-1}
>
  <div
    className="modal-dialog modal-dialog-scrollable modal-dialog-centered"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          
        </h5>
        <button
          className="close"
          onClick={[MockFunction]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`Snapshot test hidden modal 1`] = `<div />`;
